src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts

```diff
@@ -209,22 +209,32 @@ export function* calculatePageOrderAndMoveToNextPageSaga({ payload: { runValidat
     const state: IRuntimeState = yield select();
     const layoutSets = state.formLayout.layoutsets;
     const currentView = state.formLayout.uiConfig.currentView;
-    const instance = state.instanceData.instance;
-    const dataTypeId: string = getDataTaskDataTypeId(instance.process.currentTask.elementId,
-      state.applicationMetadata.applicationMetadata.dataTypes);
     let layoutSetId: string = null;
-    if (layoutSets != null) {
-      layoutSetId = getLayoutsetForDataElement(instance, dataTypeId, layoutSets);
-    }
+    let dataTypeId: string = null;
     const formData: any = convertDataBindingToModel(state.formData.formData);
+    const appIsStateless: boolean = isStatelessApp(state.applicationMetadata.applicationMetadata)
+    if (appIsStateless) {
+      dataTypeId = getCurrentDataTypeForApplication({
+        application: state.applicationMetadata.applicationMetadata,
+        layoutSets: state.formLayout.layoutsets,
+      });
+      layoutSetId = state.applicationMetadata.applicationMetadata?.onEntry?.show;
+    } else {
+      const instance = state.instanceData.instance;
+      dataTypeId = getDataTaskDataTypeId(instance.process.currentTask.elementId,
+        state.applicationMetadata.applicationMetadata.dataTypes);
+      if (layoutSets != null) {
+        layoutSetId = getLayoutsetForDataElement(instance, dataTypeId, layoutSets);
+      }
+    }
     const layoutOrder = yield call(
       post,
-      getCalculatePageOrderUrl(),
+      getCalculatePageOrderUrl(appIsStateless),
```

Comment:
Ref. #52, is this endpoint also set up with AllowAnonymous?

Reply:
just removed the authorize attribute from stateless endpoint. As the app does not store any data and the endpoint should only return the order of pages we might not need an autorized endpoint for this?